[EmitC Conversion] failure for no input ConstEval function/wrapper

EmitC generates incorrect code when const-eval function does not require any input. 

TTIR graph
```
func.func @const_eval_no_input(%arg0: tensor<71x4x1xbf16> {ttcore.argument_type = #ttcore.argument_type<input>}) -> tensor<71x4x1xbf16> {
    %0 = "ttir.full"() <{fill_value = 32.0 : f32, shape = array<i32>}> : () -> tensor<bf16>
    %1 = ttir.empty() : tensor<1x1xbf16>
    %2 = "ttir.reshape"(%0, %1) <{shape = [1 : i32, 1 : i32]}> : (tensor<bf16>, tensor<1x1xbf16>) -> tensor<1x1xbf16>
    %3 = ttir.empty() : tensor<71x4xbf16>
    %4 = "ttir.repeat"(%2, %3) <{repeat_dimensions = array<i64: 71, 4>}> : (tensor<1x1xbf16>, tensor<71x4xbf16>) -> tensor<71x4xbf16>
    %5 = ttir.empty() : tensor<71x4x1xbf16>
    %6 = "ttir.reshape"(%4, %5) <{shape = [71 : i32, 4 : i32, 1 : i32]}> : (tensor<71x4xbf16>, tensor<71x4x1xbf16>) -> tensor<71x4x1xbf16>
    %7 = ttir.empty() : tensor<71x4x1xbf16>
    %8 = "ttir.multiply"(%6, %arg0, %7) : (tensor<71x4x1xbf16>, tensor<71x4x1xbf16>, tensor<71x4x1xbf16>) -> tensor<71x4x1xbf16>
    return %8 : tensor<71x4x1xbf16>
  }
```

Generated TTNN graph; const-eval function `const_eval_no_input_const_eval_0` does not have any argument.
```
#dram = #ttnn.buffer_type<dram>
#system_desc = #ttcore.system_desc<[{role = host, target_triple = "x86_64-pc-linux-gnu"}], [{arch = <wormhole_b0>, grid = 8x8, coord_translation_offsets = 18x18, l1_size = 1499136, num_dram_channels = 12, dram_channel_size = 1073741824, noc_l1_address_align_bytes = 16, pcie_address_align_bytes = 32, noc_dram_address_align_bytes = 32, l1_unreserved_base = 1024, erisc_l1_unreserved_base = 1024, dram_unreserved_base = 1024, dram_unreserved_end = 1073741824, supported_data_types = [<f32>, <f16>, <bf16>, <bfp_f8>, <bfp_bf8>, <bfp_f4>, <bfp_bf4>, <bfp_f2>, <bfp_bf2>, <u32>, <u16>, <u8>, <si32>], supported_tile_sizes = [ 4x16,  16x16,  32x16,  4x32,  16x32,  32x32], dst_physical_size_tiles = 16, num_cbs = 32, num_compute_threads = 1, num_datamovement_threads = 2}], [0], [1 : i32], [ 0x0x0x0]>
#ttnn_layout = #ttnn.ttnn_layout<(d0, d1, d2) -> (d0 * 32 + d1, d2), <1x1>, memref<71x1x!ttcore.tile<32x32, bf16>, #dram>, <interleaved>>
#ttnn_layout1 = #ttnn.ttnn_layout<() -> (0, 0), <1x1>, memref<1x1x!ttcore.tile<32x32, bf16>, #dram>, <interleaved>>
#ttnn_layout2 = #ttnn.ttnn_layout<(d0, d1) -> (d0, d1), <1x1>, memref<1x1x!ttcore.tile<32x32, bf16>, #dram>, <interleaved>>
#ttnn_layout3 = #ttnn.ttnn_layout<(d0, d1) -> (d0, d1), <1x1>, memref<3x1x!ttcore.tile<32x32, bf16>, #dram>, <interleaved>>
module {
  ttcore.device_module {
    builtin.module attributes {ttcore.system_desc = #system_desc} {
      ttcore.device @default_device = <workerGrid = #ttcore.grid<8x8, (d0, d1) -> (0, d0, d1)>, l1Map = (d0, d1, d2)[s0] -> (0, d0, d1, d2 + s0), dramMap = (d0, d1, d2)[s0, s1, s2, s3, s4, s5, s6] -> (0, 0, (((d0 * s1) * (s2 * (s3 * s6)) + d1 * (s2 * (s3 * s6)) + d2) floordiv s4) mod 12, ((((d0 * s1) * (s2 * (s3 * s6)) + d1 * (s2 * (s3 * s6)) + d2) floordiv s4) floordiv 12) * s4 + ((d0 * s1) * (s2 * (s3 * s6)) + d1 * (s2 * (s3 * s6)) + d2) mod s4 + s5), meshShape = , chipIds = [0]>
      func.func @const_eval_no_input_const_eval_0() -> tensor<71x4x1xbf16, #ttnn_layout> attributes {const_eval} {
        %0 = "ttnn.get_device"() <{mesh_offset = #ttnn<mesh_offset 0x0>, mesh_shape = #ttnn<mesh_shape 1x1>}> : () -> !ttnn.device
        %1 = "ttnn.full"(%0) <{dtype = #ttcore.supportedDataTypes<bf16>, fill_value = 3.200000e+01 : f32, layout = #ttnn.layout<tile>, shape = #ttnn.shape<>}> : (!ttnn.device) -> tensor<bf16, #ttnn_layout1>
        %2 = "ttnn.reshape"(%1) <{shape = [1 : i32, 1 : i32]}> : (tensor<bf16, #ttnn_layout1>) -> tensor<1x1xbf16, #ttnn_layout2>
        "ttnn.deallocate"(%1) <{force = false}> : (tensor<bf16, #ttnn_layout1>) -> ()
        %3 = "ttnn.repeat"(%2) <{repeat_dims = #ttnn.shape<71x4>}> : (tensor<1x1xbf16, #ttnn_layout2>) -> tensor<71x4xbf16, #ttnn_layout3>
        "ttnn.deallocate"(%2) <{force = false}> : (tensor<1x1xbf16, #ttnn_layout2>) -> ()
        %4 = "ttnn.reshape"(%3) <{shape = [71 : i32, 4 : i32, 1 : i32]}> : (tensor<71x4xbf16, #ttnn_layout3>) -> tensor<71x4x1xbf16, #ttnn_layout>
        "ttnn.deallocate"(%3) <{force = false}> : (tensor<71x4xbf16, #ttnn_layout3>) -> ()
        return %4 : tensor<71x4x1xbf16, #ttnn_layout>
      }
      func.func @const_eval_no_input(%arg0: tensor<71x4x1xbf16, #ttnn_layout> {ttcore.argument_type = #ttcore.argument_type<input>}) -> tensor<71x4x1xbf16, #ttnn_layout> {
        %0 = ttcore.load_cached(@const_eval_no_input_const_eval_0, []) : () -> tensor<71x4x1xbf16, #ttnn_layout>
        %1 = "ttnn.multiply"(%0, %arg0) <{dtype = #ttcore.supportedDataTypes<bf16>}> : (tensor<71x4x1xbf16, #ttnn_layout>, tensor<71x4x1xbf16, #ttnn_layout>) -> tensor<71x4x1xbf16, #ttnn_layout>
        "ttnn.deallocate"(%0) <{force = false}> : (tensor<71x4x1xbf16, #ttnn_layout>) -> ()
        "ttnn.deallocate"(%arg0) <{force = false}> : (tensor<71x4x1xbf16, #ttnn_layout>) -> ()
        return %1 : tensor<71x4x1xbf16, #ttnn_layout>
      }
    }
  }
}
```

Generated C++ code
```
#include "ttnn-precompiled.hpp"
::std::vector<::ttnn::Tensor> forward_const_eval_0() {
  ttnn::distributed::MeshDevice* v1 = ttnn::DeviceGetter::getInstance();
  ::std::optional<::std::reference_wrapper<::ttnn::distributed::MeshDevice>> v2 = *v1;
  ::ttnn::Tensor v3 = ttnn::full(::ttnn::Shape({}), 32.000000f, ::ttnn::DataType::BFLOAT16, ::ttnn::Layout::TILE, v2, ::ttnn::MemoryConfig{::ttnn::TensorMemoryLayout::INTERLEAVED, ::ttnn::BufferType::DRAM, ::std::nullopt});
  ::ttnn::Tensor v4 = ttnn::reshape(v3, ::std::vector<int32_t>{1, 1}, ::ttnn::MemoryConfig{::ttnn::TensorMemoryLayout::INTERLEAVED, ::ttnn::BufferType::DRAM, ::std::nullopt});
  ttnn::deallocate(v3, false);
  ::ttnn::Tensor v5 = ttnn::repeat(v4, ::ttnn::Shape({71, 4}));
  ttnn::deallocate(v4, false);
  ::ttnn::Tensor v6 = ttnn::reshape(v5, ::std::vector<int32_t>{71, 4, 1}, ::ttnn::MemoryConfig{::ttnn::TensorMemoryLayout::INTERLEAVED, ::ttnn::BufferType::DRAM, ::std::nullopt});
  ttnn::deallocate(v5, false);
  ::std::vector<::ttnn::Tensor> v7 = util_create_vec(v6);
  return v7;
}

static ::std::vector<::ttnn::Tensor> g_cached_result_forward_const_eval_0;
::std::vector<::ttnn::Tensor> forward(::std::vector<::ttnn::Tensor> v1) {
  ::ttnn::Tensor v2 = v1[0];
  ::std::function<::std::vector<::ttnn::Tensor>(::std::vector<::ttnn::Tensor>)> v3 = &forward_const_eval_0;
  ::std::vector<::ttnn::Tensor> v4 = util_create_vec();
  ::std::vector<::ttnn::Tensor>* v5 = &g_cached_result_forward_const_eval_0;
  ttnn::constEvalFuncWrapper(v3, v4, v5);
  ::std::vector<::ttnn::Tensor> v6 = g_cached_result_forward_const_eval_0;
  ::ttnn::Tensor v7 = v6[0];
  ::ttnn::Tensor v8 = ttnn::multiply(v7, v2, ::ttnn::DataType::BFLOAT16, ::ttnn::MemoryConfig{::ttnn::TensorMemoryLayout::INTERLEAVED, ::ttnn::BufferType::DRAM, ::std::nullopt});
  ttnn::deallocate(v7, false);
  ttnn::deallocate(v2, false);
  ::std::vector<::ttnn::Tensor> v9 = util_create_vec(v8);
  return v9;
}

::std::vector<::ttnn::Tensor> create_inputs_for_forward() {
  ttnn::distributed::MeshDevice* v1 = ttnn::DeviceGetter::getInstance();
  ::std::optional<::std::reference_wrapper<::ttnn::distributed::MeshDevice>> v2 = *v1;
  ::ttnn::Tensor v3 = ttnn::ones(::ttnn::Shape({71, 4, 1}), ::ttnn::DataType::BFLOAT16, ::ttnn::Layout::TILE, v2, ::ttnn::MemoryConfig{::ttnn::TensorMemoryLayout::INTERLEAVED, ::ttnn::BufferType::DRAM, ::std::nullopt});
  ::std::vector<::ttnn::Tensor> v4 = util_create_vec(v3);
  return v4;
}

int32_t main() {
  ::std::vector<::ttnn::Tensor> v1 = create_inputs_for_forward();
  ::std::vector<::ttnn::Tensor> v2 = forward(v1);
  int32_t v3 = 0;
  return v3;
}
```

ConstEval function wrapper (`ttnn::constEvalFuncWrapper(v3, v4, v5);`) is wrong as ConstEval function does not accept any input in this case. 

`ttnn-standalone` fails while processing the above C++ code with error.
```
Using custom file for compilation: build/test/ttmlir/EmitC/TTNN/other/const-eval-empty.mlir.cpp

Building: const-eval-empty.mlir
  Error during build process: Command '['cmake', '--build', './tt-mlir/tools/ttnn-standalone/build', '--', 'ttnn-dylib']' returned non-zero exit status 1.

[1/2] Building CXX object CMakeFiles/ttnn-dylib.dir/ttnn-dylib.cpp.o
FAILED: [code=1] CMakeFiles/ttnn-dylib.dir/ttnn-dylib.cpp.o 
/usr/bin/clang++ -Dttnn_dylib_EXPORTS -I./tt-mlir/third_party/tt-metal/src/tt-metal/.cpmcache/reflect/f93e77475670eaeacf332927dfe8b50e3f3812e0 -I./tt-mlir/third_party/tt-metal/src/tt-metal/.cpmcache/fmt/69912fb6b71fcb1f7e5deca191a2bb4748c4e7b6/include -I./tt-mlir/third_party/tt-metal/src/tt-metal/.cpmcache/enchantum/2fb7ab238e36c101b9848892ddb6382276b65837/enchantum/include -I./tt-mlir/third_party/tt-metal/src/tt-metal/.cpmcache/boost/1359e136761ab2d10afa1c4e21086c8d824735cd/libs/core/include -I./tt-mlir/third_party/tt-metal/src/tt-metal/.cpmcache/nlohmann_json/798e0374658476027d9723eeb67a262d0f3c8308/include -I./tt-mlir/third_party/tt-metal/src/tt-metal/.cpmcache/spdlog/b1c2586bb5c35a7929362e87f62433eb68206873/include -I./tt-mlir/third_party/tt-metal/src/tt-metal/.cpmcache/tt-logger/5d45c661affa374c5656335a77010ab503f96a99/include -I./tt-mlir/third_party/tt-metal/src/tt-metal -I./tt-mlir/third_party/tt-metal/src/tt-metal/tt_metal -I./tt-mlir/third_party/tt-metal/src/tt-metal/tt_metal/api -I./tt-mlir/third_party/tt-metal/src/tt-metal/tt_metal/hostdevcommon/api -I./tt-mlir/third_party/tt-metal/src/tt-metal/tt_metal/hw/inc -I./tt-mlir/third_party/tt-metal/src/tt-metal/tt_metal/hw/inc/grayskull -I./tt-mlir/third_party/tt-metal/src/tt-metal/tt_metal/hw/inc/wormhole/wormhole_b0_defines -I./tt-mlir/third_party/tt-metal/src/tt-metal/tt_metal/hw/inc/wormhole -I./tt-mlir/third_party/tt-metal/src/tt-metal/tt_metal/hw/inc/blackhole -I./tt-mlir/third_party/tt-metal/src/tt-metal/tt_metal/include -I./tt-mlir/third_party/tt-metal/src/tt-metal/tt_stl -I./tt-mlir/third_party/tt-metal/src/tt-metal/tt_stl/tt_stl -I./tt-mlir/third_party/tt-metal/src/tt-metal/tt_metal/third_party/fmt -I./tt-mlir/third_party/tt-metal/src/tt-metal/tt_metal/third_party/enchantum -I./tt-mlir/third_party/tt-metal/src/tt-metal/tt_metal/third_party/taskflow -I./tt-mlir/third_party/tt-metal/src/tt-metal/tt_metal/third_party/tracy/public -I./tt-mlir/third_party/tt-metal/src/tt-metal/tt_metal/third_party/umd -I./tt-mlir/third_party/tt-metal/src/tt-metal/tt_metal/third_party/umd/device/api -I./tt-mlir/third_party/tt-metal/src/tt-metal/tt_metal/third_party/umd/src/firmware/riscv/grayskull -I./tt-mlir/third_party/tt-metal/src/tt-metal/tt_metal/third_party/umd/src/firmware/riscv/wormhole -I./tt-mlir/third_party/tt-metal/src/tt-metal/tt_metal/third_party/umd/src/firmware/riscv/blackhole -I./tt-mlir/third_party/tt-metal/src/tt-metal/ttnn -I./tt-mlir/third_party/tt-metal/src/tt-metal/ttnn/api -I./tt-mlir/third_party/tt-metal/src/tt-metal/ttnn/core -I./tt-mlir/third_party/tt-metal/src/tt-metal/ttnn/cpp -I./tt-mlir/third_party/tt-metal/src/tt-metal/ttnn/cpp/ttnn -O3 -DNDEBUG -std=gnu++20 -fPIC -march=x86-64-v3 -Winvalid-pch -Xclang -include-pch -Xclang ./tt-mlir/tools/ttnn-standalone/build/CMakeFiles/ttnn-dylib.dir/cmake_pch.hxx.pch -Xclang -include -Xclang ./tt-mlir/tools/ttnn-standalone/build/CMakeFiles/ttnn-dylib.dir/cmake_pch.hxx -MD -MT CMakeFiles/ttnn-dylib.dir/ttnn-dylib.cpp.o -MF CMakeFiles/ttnn-dylib.dir/ttnn-dylib.cpp.o.d -o CMakeFiles/ttnn-dylib.dir/ttnn-dylib.cpp.o -c ./tt-mlir/tools/ttnn-standalone/ttnn-dylib.cpp
./tt-mlir/tools/ttnn-standalone/ttnn-dylib.cpp:19:81: error: no viable conversion from '::std::vector< ::ttnn::Tensor> (*)()' (aka 'vector<tt::tt_metal::Tensor> (*)()') to '::std::function< ::std::vector< ::ttnn::Tensor> (::std::vector< ::ttnn::Tensor>)>' (aka 'function<vector<tt::tt_metal::Tensor> (vector<tt::tt_metal::Tensor>)>')
   19 |   ::std::function<::std::vector<::ttnn::Tensor>(::std::vector<::ttnn::Tensor>)> v3 = &forward_const_eval_0;
      |                                                                                 ^    ~~~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/std_function.h:375:7: note: candidate constructor not viable: no known conversion from '::std::vector< ::ttnn::Tensor> (*)()' (aka 'vector<tt::tt_metal::Tensor> (*)()') to 'nullptr_t' (aka 'std::nullptr_t') for 1st argument
  375 |       function(nullptr_t) noexcept
      |       ^        ~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/std_function.h:386:7: note: candidate constructor not viable: no known conversion from '::std::vector< ::ttnn::Tensor> (*)()' (aka 'vector<tt::tt_metal::Tensor> (*)()') to 'const function<vector<Tensor> (vector<Tensor>)> &' for 1st argument
  386 |       function(const function& __x)
      |       ^        ~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/std_function.h:404:7: note: candidate constructor not viable: no known conversion from '::std::vector< ::ttnn::Tensor> (*)()' (aka 'vector<tt::tt_metal::Tensor> (*)()') to 'function<vector<Tensor> (vector<Tensor>)> &&' for 1st argument
  404 |       function(function&& __x) noexcept
      |       ^        ~~~~~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/std_function.h:435:2: note: candidate template ignored: requirement '_Callable<std::vector<tt::tt_metal::Tensor, std::allocator<tt::tt_metal::Tensor>> (*)(), std::vector<tt::tt_metal::Tensor, std::allocator<tt::tt_metal::Tensor>> (*)(), std::__invoke_result<std::vector<tt::tt_metal::Tensor, std::allocator<tt::tt_metal::Tensor>> (*&)(), std::vector<tt::tt_metal::Tensor, std::allocator<tt::tt_metal::Tensor>>>>::value' was not satisfied [with _Functor = ::std::vector< ::ttnn::Tensor> (*)()]
  435 |         function(_Functor&& __f)
      |         ^
./tt-mlir/tools/ttnn-standalone/ttnn-dylib.cpp:22:35: error: no viable conversion from 'function<::std::vector< ::ttnn::Tensor> (::std::vector< ::ttnn::Tensor>)>' to 'function<std::vector<ttnn::Tensor> ()>'
   22 |   ttnn::constEvalFuncWrapperEmpty(v3, v5);
      |                                   ^~
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/std_function.h:375:7: note: candidate constructor not viable: no known conversion from '::std::function< ::std::vector< ::ttnn::Tensor> (::std::vector< ::ttnn::Tensor>)>' (aka 'function<vector<tt::tt_metal::Tensor> (vector<tt::tt_metal::Tensor>)>') to 'nullptr_t' (aka 'std::nullptr_t') for 1st argument
  375 |       function(nullptr_t) noexcept
      |       ^        ~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/std_function.h:386:7: note: candidate constructor not viable: no known conversion from '::std::function< ::std::vector< ::ttnn::Tensor> (::std::vector< ::ttnn::Tensor>)>' (aka 'function<vector<tt::tt_metal::Tensor> (vector<tt::tt_metal::Tensor>)>') to 'const function<vector<Tensor> ()> &' for 1st argument
  386 |       function(const function& __x)
      |       ^        ~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/std_function.h:404:7: note: candidate constructor not viable: no known conversion from '::std::function< ::std::vector< ::ttnn::Tensor> (::std::vector< ::ttnn::Tensor>)>' (aka 'function<vector<tt::tt_metal::Tensor> (vector<tt::tt_metal::Tensor>)>') to 'function<vector<Tensor> ()> &&' for 1st argument
  404 |       function(function&& __x) noexcept
      |       ^        ~~~~~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/bits/std_function.h:435:2: note: candidate template ignored: requirement '_Callable<std::function<std::vector<tt::tt_metal::Tensor, std::allocator<tt::tt_metal::Tensor>> (std::vector<tt::tt_metal::Tensor, std::allocator<tt::tt_metal::Tensor>>)> &, std::function<std::vector<tt::tt_metal::Tensor, std::allocator<tt::tt_metal::Tensor>> (std::vector<tt::tt_metal::Tensor, std::allocator<tt::tt_metal::Tensor>>)>, std::__invoke_result<std::function<std::vector<tt::tt_metal::Tensor, std::allocator<tt::tt_metal::Tensor>> (std::vector<tt::tt_metal::Tensor, std::allocator<tt::tt_metal::Tensor>>)> &>>::value' was not satisfied [with _Functor = ::std::function< ::std::vector< ::ttnn::Tensor> (::std::vector< ::ttnn::Tensor>)> &]
  435 |         function(_Functor&& __f)
      |         ^
./tt-mlir/tools/ttnn-standalone/ttnn-precompiled.hpp:147:9: note: passing argument to parameter 'constEvalFunc' here
  147 |         constEvalFunc,
      |         ^
2 errors generated.
ninja: build stopped: subcommand failed.
```

--- a/lib/Conversion/TTNNToEmitC/TTNNToEmitC.cpp
+++ b/lib/Conversion/TTNNToEmitC/TTNNToEmitC.cpp
@@ -2022,10 +2022,17 @@ public:
     // Restore the insertion point to continue with the function
     rewriter.restoreInsertionPoint(currentInsertionPoint);
 
+    const bool isZeroArgWrapper = adaptor.getInputs().size() == 0;
     // Create the function pointer type
-    auto funcPtrType = emitc::OpaqueType::get(
-        rewriter.getContext(), "::std::function<::std::vector<::ttnn::Tensor>(:"
-                               ":std::vector<::ttnn::Tensor>)>");
+    auto funcPtrType =
+        isZeroArgWrapper
+            ? emitc::OpaqueType::get(
+                  rewriter.getContext(),
+                  "::std::function<::std::vector<::ttnn::Tensor>()>")
+            : emitc::OpaqueType::get(
+                  rewriter.getContext(),
+                  "::std::function<::std::vector<::ttnn::Tensor>(:"
+                  ":std::vector<::ttnn::Tensor>)>");
     auto addressAttr =
         emitc::OpaqueAttr::get(rewriter.getContext(), "&" + callee.str());
     auto funcPtrValue = rewriter.create<emitc::ConstantOp>(
@@ -2049,9 +2056,15 @@ public:
                                                        "&", globalVar);
 
     // Call the wrapper function with the pointer
-    rewriter.create<emitc::CallOpaqueOp>(
-        srcOp.getLoc(), TypeRange{}, "ttnn::constEvalFuncWrapper",
-        ValueRange{funcPtrValue, tupleValue, addressOfOp}, ArrayAttr{});
+    if (isZeroArgWrapper) {
+      rewriter.create<emitc::CallOpaqueOp>(
+          srcOp.getLoc(), TypeRange{}, "ttnn::constEvalFuncWrapperZeroArg",
+          ValueRange{funcPtrValue, addressOfOp}, ArrayAttr{});
+    } else {
+      rewriter.create<emitc::CallOpaqueOp>(
+          srcOp.getLoc(), TypeRange{}, "ttnn::constEvalFuncWrapper",
+          ValueRange{funcPtrValue, tupleValue, addressOfOp}, ArrayAttr{});
+    }
 
     // Load the value from the global variable
     auto resultVar =

--- a/test/ttmlir/EmitC/TTNN/other/const-eval.mlir
+++ b/test/ttmlir/EmitC/TTNN/other/const-eval.mlir
@@ -21,4 +21,18 @@ module {
     %9 = "ttir.multiply"(%1, %7, %8) : (tensor<32x32xbf16>, tensor<32x32xbf16>, tensor<32x32xbf16>) -> tensor<32x32xbf16>
     return %9 : tensor<32x32xbf16>
   }
+
+  func.func @const_eval_no_input(%arg0: tensor<71x4x1xbf16> {ttcore.argument_type = #ttcore.argument_type<input>}) -> tensor<71x4x1xbf16> {
+    // CHECK: = ttcore.load_cached(@const_eval_no_input_const_eval_0, [])
+    %0 = "ttir.full"() <{fill_value = 32.0 : f32, shape = array<i32>}> : () -> tensor<bf16>
+    %1 = ttir.empty() : tensor<1x1xbf16>
+    %2 = "ttir.reshape"(%0, %1) <{shape = [1 : i32, 1 : i32]}> : (tensor<bf16>, tensor<1x1xbf16>) -> tensor<1x1xbf16>
+    %3 = ttir.empty() : tensor<71x4xbf16>
+    %4 = "ttir.repeat"(%2, %3) <{repeat_dimensions = array<i64: 71, 4>}> : (tensor<1x1xbf16>, tensor<71x4xbf16>) -> tensor<71x4xbf16>
+    %5 = ttir.empty() : tensor<71x4x1xbf16>
+    %6 = "ttir.reshape"(%4, %5) <{shape = [71 : i32, 4 : i32, 1 : i32]}> : (tensor<71x4xbf16>, tensor<71x4x1xbf16>) -> tensor<71x4x1xbf16>
+    %7 = ttir.empty() : tensor<71x4x1xbf16>
+    %8 = "ttir.multiply"(%6, %arg0, %7) : (tensor<71x4x1xbf16>, tensor<71x4x1xbf16>, tensor<71x4x1xbf16>) -> tensor<71x4x1xbf16>
+    return %8 : tensor<71x4x1xbf16>
+  }
 }

--- a/tools/ttnn-standalone/ttnn-precompiled.hpp
+++ b/tools/ttnn-standalone/ttnn-precompiled.hpp
@@ -126,7 +126,7 @@ void setDevice(ttnn::MeshDevice *device) { DeviceGetter::setInstance(device); }
 }
 
 // Wrapper to abstract const-eval logic out of runtime funcs to keep them
-// cleaner.  Invokes constEvalFunc iff outputs is empty.
+// cleaner. Invokes constEvalFunc iff outputs is empty.
 void constEvalFuncWrapper(
     std::function<std::vector<ttnn::Tensor>(std::vector<ttnn::Tensor>)>
         constEvalFunc,
@@ -134,6 +134,19 @@ void constEvalFuncWrapper(
     std::vector<ttnn::Tensor> *outputs) {
   if (outputs->empty()) {
     *outputs = constEvalFunc(inputs);
+  }
+}
+
+// Wrapper to abstract const-eval logic out of runtime funcs to keep them
+// cleaner. Invokes constEvalFunc iff outputs is empty.
+// This is an overload of constEvalFuncWrapper for const-eval functions that
+// take zero arguments.
+void constEvalFuncWrapperZeroArg(
+    std::function<std::vector<ttnn::Tensor>()> constEvalFunc,
+    // const std::vector<ttnn::Tensor> &inputs,
+    std::vector<ttnn::Tensor> *outputs) {
+  if (outputs->empty()) {
+    *outputs = constEvalFunc();
   }
 }
 

